include/oneapi/dpl/internal/async_impl/async_impl.h

```diff
@@ -0,0 +1,91 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) 2021 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _ONEDPL_ASYNC_IMPL_H
+#define _ONEDPL_ASYNC_IMPL_H
+
+#if _ONEDPL_HETERO_BACKEND
+#    include "async_impl_hetero.h"
+#endif
+
+#include "glue_async_impl.h"
+
+namespace oneapi
+{
+namespace dpl
+{
+namespace experimental
+{
+
+// [wait_for_all]
+template <typename... _Ts>
+typename ::std::enable_if<(true && ... && ::std::is_convertible_v<_Ts, sycl::event>), void>::type
+wait_for_all(_Ts&&... __events)
+{
+    ::std::initializer_list<int> i = {0, (__events.wait(), 0)...};
+    (void)i;
```

Comment:
Do we create an initialization list just for call `wait` for the all events?
I think explicit calls `wait` are more readable..

Reply:
Please enlighten me. How would you further simplify member function calls on the parameter pack?

Reply:
Here you are
https://godbolt.org/z/Ynznaf

Reply:
Another question  - whether compiler remove variable `::std::initializer_list<int> i` as unused?